PROTO_tests/tests/join_test.py

```diff
@@ -0,0 +1,226 @@
+import numpy as np
+import pytest
+
+import arkouda as ak
+
+"""
+Encapsulates a variety of arkouda join_on_eq_with_dt test cases.
+"""
+
+
+class TestJoin:
+    @classmethod
+    def setup_class(cls):
+        cls.size = 1000
+        cls.a1 = ak.ones(cls.size, dtype=np.int64)
+        cls.a2 = ak.arange(0, cls.size, 1)
+        cls.t1 = cls.a1
+        cls.t2 = cls.a1 * 10
+        cls.dt = 10
+        ak.verbose = False
+
+    @pytest.mark.parametrize("dt_type", ["true_dt", "abs_dt", "pos_dt"])
+    def test_join_on_eq_by_dt(self, dt_type):
+        x, y = ak.join_on_eq_with_dt(self.a2, self.a1, self.t1, self.t2, self.dt, dt_type)
+        nl = ak.get_config()["numLocales"]
+        assert self.size // nl, x.size
+        assert self.size // nl, y.size
+
+    def test_join_on_eq_with_true_dt_with_result_limit(self):
+        nl = ak.get_config()["numLocales"]
+        lim = (self.size + nl) * self.size
+        res_size = self.size * self.size
+        x, y = ak.join_on_eq_with_dt(
+            self.a1, self.a1, self.a1, self.a1, self.dt, "true_dt", result_limit=lim
+        )
+        assert res_size, x.size
+        assert res_size, y.size
+
+    def test_join_on_eq_with_abs_dt_outside_window(self):
+        """
+        Should get 0 answers because N^2 matches but 0 within dt window
+        """
+        dt = 8
+        x, y = ak.join_on_eq_with_dt(self.a1, self.a1, self.t1, self.t1 * 10, dt, "abs_dt")
+        assert 0 == x.size
+        assert 0 == y.size
+
+        x, y = ak.join_on_eq_with_dt(self.a2, self.a1, self.t1, self.t2, dt, "abs_dt")
+        assert 0 == x.size
+        assert 0 == y.size
```

Comment:
feels like we could reduce some code by looping here and substituting the third from last argument

Reply:
I've done this in a couple other spots.  Here, both the first and fourth args are different, so I haven't (yet) made this change.  Please review the loops added in the latest commit and let me know your thoughts on them and this one.

Reply:
Yah I'm not sure it's worth it, let's hold off on it